src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -348,6 +366,26 @@ export class HubConnection {
         }
     }
 
+    /** Registers a handler that will be invoked when the connection starts reconnecting.
+     *
+     * @param {Function} callback The handler that will be invoked when the connection start reconnecting. Optionally receives a single argument containing the error that caused the connection to start reconnecting (if any).
+     */
+    public onreconnecting(callback: (error?: Error) => void) {
+        if (callback) {
+            this.reconnectingCallbacks.push(callback);
```

Comment:
We might just want this to be a single handler that you can overwrite by calling this function again.

If we allow multiple registrations then people will want to be able to unregister.

Reply:
I'm following the pattern we used for onclose(). I think if we want to change that pattern, we can change it for all onclose/onreconnecting/onreconnected all at once.